### ISSUE TITLE
[Feature]: a11y: support WCAG 2.2 by not having "hidden controls" in DetailsList

### ISSUE THREAD

**Issue by dgreene1** [microsoft/fluentui#23667]
### Library

React / v8 (@fluentui/react)

### Describe the feature that you would like added

##Context

The [visible controls (formerly hidden controls) section of the draft 2.2](https://w3c.github.io/wcag/understanding/visible-controls.html) spec states:

> Success Criterion [3.2.7 Visible Controls](https://w3c.github.io/wcag/guidelines/#visible-controls) (Level AA): When [user interface components](https://w3c.github.io/wcag/understanding/visible-controls.html#dfn-user-interface-component) are invisible until hover or focus makes them visible, provide a visible indicator that the components are available, except when:
> - The same functionality is available through a component on the same page, or on a different step in a multi-step process,  without requiring pointer hover or keyboard focus to make it visible;
> - The component provides keyboard-only functionality;
> - A mechanism is available to make the components persistently visible;
> - Hiding the visual indicator is [essential](https://w3c.github.io/wcag/understanding/visible-controls.html#dfn-essential).

## Issue

So when you look at [DetailedList example](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/grouped) you see that the div with class "ms-Check" is hidden unless the user hovers over the row.

## Proposal

* The "ms-Check" div for all rows would be visible at all times.
  * When all of the items in the group are selected, it would show as fully checked
  * When some of the items in the group are selected, it would show as "mixed" checked state (ala [aria-checked](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role#associated_wai-aria_roles_states_and_properties)'s 3 states)
  * When none of the items in the group are selected it would shown as unchecked
* The "ms-Check" div in the table header would also be visible at all times and would follow the [aria-checked true/false/mixed rules](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role#associated_wai-aria_roles_states_and_properties) as well (that I enumerated above) 

### Have you discussed this feature with our team

I have not. FYI, my team and I are willing to implement this feature. 💪 

### Additional context

As you can see, only one of the checkboxes is visible until you hover your mouse:
![image](https://user-images.githubusercontent.com/14020024/175117606-11c0d563-0dcf-47c7-ba04-d1042b29433b.png)

This is particularly strange when you select all children, and then fold the group... because it doesn't show you any selection information at all.

See this gif below:
![2022-06-22_15-14-19](https://user-images.githubusercontent.com/14020024/175118155-11aa8b34-2092-4db8-8284-3223557f3b14.gif)

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by dgreene1** [comment]
I want to emphasize the point that my team and I can implement this feature. Given that not everyone might like this style, we could make it an optional parameter like `alwaysShowSelectIndicator`. Anyone who doesn't set this would be failing the [Draft WCAG 2.2](https://w3c.github.io/wcag/understanding/visible-controls.html), but given that it's draft I think an optional parameter allows us to continue as if it was simply a style preference.
